### PR TITLE
Add test for dict encoded string types - omniscidb-internal PR 5492

### PR DESCRIPTION
**Pull request by guilhermeleobas**
As title. Requires PR https://github.com/omnisci/omniscidb-internal/pull/5492

### CODE REVIEW

File: rbc/tests/test_omnisci_shared_dictionary.py
Comment (guilhermeleobas) [inline rbc/tests/test_omnisci_shared_dictionary.py:33]: Note for myself: wait until omnisci/omniscidb-internal#5492 gets merged and make sure this works with the omnisci-latest image we have on CI.

File: rbc/tests/__init__.py
Comment (guilhermeleobas) [inline rbc/tests/__init__.py:71]: We don't use TRAVIS CI anymore.

**Comment by guilhermeleobas** [comment]
Failures on omniscidb v5.5 and v5.4 are real and I need to investigate why they happen.

File: rbc/omnisci_backend/omnisci_column.py
Comment (guilhermeleobas) [inline rbc/omnisci_backend/omnisci_column.py:28]: Note for the future: wait until omnisci/omniscidb-internal#5492 lands in omniscidb dev image to revert this to (`5, 7)`

File: rbc/remotejit.py
Comment (guilhermeleobas) [inline rbc/remotejit.py:328]: This line is required for `print(a_table_function)` to work. Otherwise, the typesystem will raise a [TypeError](https://github.com/xnd-project/rbc/blob/59dcb06ccf8fcfffbd2c0d4e733b67d01062a94c/rbc/typesystem.py#L1536)

**Review by pearu — APPROVED** [review]
LGTM

**Pull request merged**
